presto-hive/src/main/java/io/prestosql/plugin/hive/metastore/glue/GlueColumnStatisticsProvider.java

```diff
@@ -32,7 +30,7 @@
 
     Map<String, HiveColumnStatistics> getPartitionColumnStatistics(Partition partition);
 
-    void updateTableColumnStatistics(TableInput table, Map<String, HiveColumnStatistics> columnStatistics);
+    void updateTableColumnStatistics(Table table, Map<String, HiveColumnStatistics> columnStatistics);
 
-    void updatePartitionStatistics(PartitionInput partition, Map<String, HiveColumnStatistics> columnStatistics);
+    void updatePartitionStatistics(Partition partition, Map<String, HiveColumnStatistics> columnStatistics, boolean forPartitionCreation);
```

Comment:
Does the change suggested in `updateTableColumnStatistics` allow us to remove this `boolean` control variable?

I'm not in favour of using boolean arguments to decide method behaviour.

Reply:
Totally agree with removing the boolean flag, the proposed change should address this